### PR TITLE
Remove excess memory allocation

### PR DESCRIPTION
**Pull request by wzshiming**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

There is no need to alloc memory for Msg-data every time in the loop.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/stream/edgedexecconnection.go [hunk 1]
```diff
@@ -93,16 +93,15 @@ func (e *EdgedExecConnection) Serve(tunnel SafeWriteTunneler) error {
...
 		select {
 		case <-stop:
 			klog.V(6).Infof("receive stop single, so stop exec scan ...")
 			return nil
 		default:
 		}
-		data := make([]byte, 256)
-
-		n, err := con.Read(data)
+		n, err := con.Read(data[:])
```
Comment (daixiang0) [inline pkg/stream/edgedexecconnection.go:104]: ?

Comment (wzshiming) [reply pkg/stream/edgedexecconnection.go:104]: There is no need to alloc memory for Msg-data every time in the loop. always use `var data [256]byte`

Comment (daixiang0) [reply pkg/stream/edgedexecconnection.go:104]: Why [:]?

Comment (wzshiming) [reply pkg/stream/edgedexecconnection.go:104]: Reduce small memory allocation and reduce GC time

Comment (wzshiming) [reply pkg/stream/edgedexecconnection.go:104]: data is an Array type, not a Slice, need  [:] as to slice.

Comment (fisherxu) [reply pkg/stream/edgedexecconnection.go:104]: Thanks for you fixing! :) And I'm not sure whether `con.Read(data[:])`  will refresh the whole byte in `data` every time? Or there wil be dirty byte in `data`. @wzshiming

Comment (zc2638) [reply pkg/stream/edgedexecconnection.go:104]: This is reference passing, similar to pointer passing @fisherxu

Comment (wzshiming) [reply pkg/stream/edgedexecconnection.go:104]: @fisherxu 
`n, err := con.Read(data[:])`, will write `n` length data in `data[:]`, below use `data[:n]`.

Comment (fisherxu) [reply pkg/stream/edgedexecconnection.go:104]: Make sense.

**Pull request merged**